doc/api/module.md

```diff
@@ -693,8 +703,24 @@ register('./my-hooks.mjs', {
 });
 ```
 
+### In-thread Synchronous Customization Hooks
```

Comment:
> I wouldn’t want someone seeing register as more mature and more stable than registerHooks and therefore a safer choice.

I think that is an accurate description for now, considering `module.registerHooks()` is not yet even released, and `module.register()` has already been used by several projects. The future migration path should be a follow up after we actually get some feedback from the user land. This is a semver-minor PR and we should keep it so. The problem of the off-thread hooks came from dictating a path before sufficient real-world feedback was collected, and I don't think we should make the same mistake again just because we believe "this time, we got it right!" (I hope we are, but this isn't up to us).

Reply:
If you don’t want to include anything about “we consider this to be the way forward” just yet, fine, but I still think that the new hooks should go first; and the explanation about how the hooks compare with each other should live with the async hooks rather than with the sync hooks. That way when we remove the async hooks, the comparison text is already there and goes away cleanly.

Reply:
I spent some time to move it and it took way to much effort to do an overhaul of the texts to describe the sync hooks first, because the existing texts always assume they are talking about the async ones and need wording changes everywhere. I don't think this needs to be done in the initial PR that's mostly just implementing things. Can you open a follow-up PR to do the rewrite?